### PR TITLE
Add v6 MIGRATION.md

### PR DESCRIPTION
**Pull request by notlee**
I'm adding to the checklist tasks to:
- update README
- correct SassDoc
- add sass tests

### CODE REVIEW

**Comment by origamiserviceuser** [comment]
o-buttons bundle size difference from 5.16.7 to f8d9d61d05b21019f8baa83b744495cfab8a37bd
css, master: 23.35kb decrease (-0.49kb/gzip)
css, internal: 27.27kb decrease (-0.51kb/gzip)
css, whitelabel: 88.77kb decrease (-2.32kb/gzip)

**Comment by origamiserviceuser** [comment]
o-buttons bundle size difference from 5.16.7 to d6f7b0a8ef2e3869e2e755ea04e8c10635dc3625
css, master: 23.35kb decrease (-0.49kb/gzip)
css, internal: 27.27kb decrease (-0.51kb/gzip)
css, whitelabel: 88.77kb decrease (-2.32kb/gzip)

File: MIGRATION.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
+## Migration Guide
```
Comment (chee) [inline MIGRATION.md]: Are the migration guides meant to start at h2?

Comment (notlee) [reply MIGRATION.md]: They all do but... why

**Pull request merged**